restAPI/restApi_test.go

```diff
@@ -56,16 +57,42 @@ func (mock *MockExtensionController) GetAllExtensions(dbConnectionWithNoAutocomm
 
 func (suite *RestAPISuite) TestGetInstallations() {
 	suite.controller.On("GetAllInstallations", mock.Anything).Return([]*extensionAPI.JsExtInstallation{{Name: "test", Version: "0.1.0", InstanceParameters: []interface{}{map[string]interface{}{"id": "param1", "name": "My param", "type": "string"}}}}, nil)
-	responseString := suite.makeGetRequest("/installations")
+	responseString := suite.makeGetRequest("/installations?dbHost=host&dbPort=8563&dbUser=user&dbPass=password")
 	suite.assertJSON.Assertf(responseString, `{"installations":[{"name":"test","version":"0.1.0","instanceParameters":[{"id":"param1","name":"My param","type":"string"}]}]}`)
 }
 
 func (suite *RestAPISuite) TestGetExtensions() {
 	suite.controller.On("GetAllExtensions", mock.Anything).Return([]*extensionController.Extension{{Name: "my-extension", Description: "a cool extension", InstallableVersions: []string{"0.1.0"}}}, nil)
-	responseString := suite.makeGetRequest("/extensions")
+	responseString := suite.makeGetRequest("/extensions?dbHost=host&dbPort=8563&dbUser=user&dbPass=password")
 	suite.assertJSON.Assertf(responseString, `{"extensions":[{"name":"my-extension","description":"a cool extension","installableVersions":["0.1.0"]}]}`)
 }
 
+func (suite *RestAPISuite) TestRequestsFailForMissingParameters() {
+	var tests = []struct {
+		url           string
+		parameters    string
+		expectedError string
+	}{
+		{"/extensions", "dbPort=8563&dbUser=user&dbPass=password", "missing dbHost"},
+		{"/extensions", "dbHost=host&dbUser=user&dbPass=password", "missing dbPort"},
+		{"/extensions", "dbHost=host&dbPort=invalidPort&dbUser=user&dbPass=password", "invalid dbPort"},
+		{"/extensions", "dbHost=host&dbPort=8563&dbPass=password", "missing dbUser"},
+		{"/extensions", "dbHost=host&dbPort=8563&dbUser=user", "missing dbPass"},
+		{"/installations", "dbPort=8563&dbUser=user&dbPass=password", "missing dbHost"},
+		{"/installations", "dbHost=host&dbUser=user&dbPass=password", "missing dbPort"},
+		{"/installations", "dbHost=host&dbPort=invalidPort&dbUser=user&dbPass=password", "invalid dbPort"},
+		{"/installations", "dbHost=host&dbPort=8563&dbPass=password", "missing dbUser"},
+		{"/installations", "dbHost=host&dbPort=8563&dbUser=user", "missing dbPass"},
+	}
+	suite.controller.On("GetAllExtensions", mock.Anything).Return([]*extensionController.Extension{{Name: "my-extension", Description: "a cool extension", InstallableVersions: []string{"0.1.0"}}}, nil)
+	suite.controller.On("GetAllInstallations", mock.Anything).Return([]*extensionAPI.JsExtInstallation{{Name: "test", Version: "0.1.0", InstanceParameters: []interface{}{map[string]interface{}{"id": "param1", "name": "My param", "type": "string"}}}}, nil)
+	for _, test := range tests {
+		completeUrl := fmt.Sprintf("%s?%s", test.url, test.parameters)
+		responseString := suite.makeGetRequest(completeUrl)
+		suite.Equal("Internal error.", responseString, fmt.Sprintf("Expected request %s to fail", completeUrl))
```

Comment:
A helpful error message must be returned here instead of "Internal error". This will be done in https://github.com/exasol/extension-manager/issues/15